src/v/security/request_auth.cc

```diff
@@ -160,6 +161,25 @@ request_auth_result request_authenticator::do_authenticate(
                   request_auth_result::superuser(superuser));
             }
         }
+    } else if (supports("OIDC") && auth_hdr.substr(0, 6) == "Bearer") {
+        // Minimal length: Bearer, a space, 1 or more bytes
+        if (auth_hdr.size() < 8) {
+            throw ss::httpd::bad_request_exception(
+              "Malformed Authorization header");
+        }
+        auto auth = security::oidc::authenticator{
+          _controller->get_oidc_service().local()};
+        auto res = auth.authenticate(auth_hdr.substr(7));
+        if (res.has_error()) {
+            throw ss::httpd::base_exception(
+              "Unauthorized", ss::http::reply::status_type::unauthorized);
+        }
+        const auto& superusers = _superusers();
+        auto found = std::find(superusers.begin(), superusers.end(), username);
```

Comment:
should username be initialized somewhere on the oidc path?

Reply:
> should username be initialized somewhere on the oidc path?

Good spot. Looks like I've dropped some tests in this area.

Reply:
This is all different, and fixed.